test/integration/scheduler/scheduler_test.go

```diff
@@ -1267,12 +1267,14 @@ var _ = ginkgo.Describe("Scheduler", func() {
 
 			ginkgo.By("Finishing second workload for prod ClusterQueue")
 			util.FinishWorkloads(ctx, k8sClient, pWl2)
+			util.ExpectWorkloadToFinish(ctx, k8sClient, client.ObjectKeyFromObject(pWl2))
 			// The pWl3 workload gets accepted, even though it was created after dWl1.
```

Comment:
Also please use `util.WaitForNextSecondAfterCreation(dWl1)` after dWL1 created and before pWl3 created to implement this case..